### ISSUE TITLE
Column Name of a Table Question with duplicate column name can't be mapped to the configured Field Name 

### ISSUE THREAD

**Issue by joboyx** [metabase/metabase#7959]
> Before filing an issue we'd appreciate it if you could take a moment to ensure
> there isn't already an open issue or pull-request.

I searched for "is:issue duplicate column name" and found nothing related.

### Bugs
If you're filing an issue about a bug please include as much information
as you can including the following.

- Your browser and the version: (67.0.3396.87 (Official Build) (64-bit))
- Your operating system: (Ubuntu 17.10)
- Your databases: (Oracle 11c)
- Metabase version: (v0.29.3)
- Metabase hosting environment: (Ubuntu 17.10)
- Metabase internal database: (MySQL 5.5)

### Steps to Reproduce
- Create a question querying fields to 2 dimension tables (join) from a fact table that has the same column_name (ex. T1.NAME and T2.NAME)
- Change the Display Name in Data Models to "T1 Name" and "T2 Name"
- Query the question grouped by the 2 fields named "NAME" and the resulting table would have the column name of the 2nd column as "Name 2" instead of "T2 Name" 

**ISSUE**: Metabase wasn't able to match the Display Name configured by admin

![image](https://user-images.githubusercontent.com/920888/42081675-9f860cfa-7bb8-11e8-8c28-4479b0dc6fa3.png)

**Comment by joboyx** [comment]
Just want to BUMP this ticket again. Since the experience is frustrating. And I can't solve my problem.

I'm encountering a similar scenario to this with the following conditions:
- Given a Question query with 2 columns from 2 different tables joined together that have column field aliases, example:
```
Group By "Group < Group Code"(`GROUP.CODE`) and Product < Product Code (`PRODUCT.CODE`) and "Member Name"(`MEMBER.NAME`)

NOTE: "Group Code" and "Product Code" are the Data Model aliases of the 2 CODE columns.
```

- If you do this group by, an unpredictable behavior happens, either:
  - sometimes the 2nd column with the same name are not displayed in the table output (i.e. `PRODUCT.CODE`)
  - sometimes the columns with same column name will be displayed, but other GROUP BY columns will not be displayed in the tables results (i.e. `MEMBER.NAME`)

----

Please let me know if the details are confusing. I hope I get my message across clearly.

**Based on this, I do think this should be at least P2 since the user can't query a specific field and there's no workaround that I know of.**

**Comment by mazameli** [comment]
@camsaul @tlrobinson Do we think this issue will also be resolved by the fix for de-duping column names?

**Comment by PauloHFM** [comment]
Getting the same error on v30.3

![image](https://user-images.githubusercontent.com/38223133/47669653-1ccd8180-db8a-11e8-8aec-55f3ba96afae.png)

**Comment by flamber** [comment]
I think this was fixed since 0.33 - I cannot reproduce in 0.35.3.
Now shows columns as "(normalized) table name -> column display name"
![image](https://user-images.githubusercontent.com/1447303/81611001-a47c8900-93da-11ea-8936-00b6480b61ac.png)
